src/http.js

```diff
@@ -71,7 +71,7 @@ class Http {
 
     let retriedOnAuthError = false;
     const execute = () => {
-      const promise = this.prepareRequest(request)
+      const promise = () => this.prepareRequest(request)
```

Comment:
Can we change the variable name?  It's now a function that returns a promise, not a promise